File: backend/schemas/order.py
from decimal import Decimal

from pydantic import BaseModel, ConfigDict


class OrderItemCreate(BaseModel):
    product_id: int
    quantity: int


class OrderCreate(BaseModel):
    customer_id: int
    items: list[OrderItemCreate]


class OrderItemResponse(BaseModel):
    id: int
    product_id: int
    quantity: int
    purchased_price: Decimal
    model_config = ConfigDict(from_attributes=True)


class OrderResponse(BaseModel):
    id: int
    customer_id: int
    items: list[OrderItemResponse]
    model_config = ConfigDict(from_attributes=True)

File: frontend/types/order.ts
import type { ApiMessageResponse, DecimalString } from "./api";

export interface OrderItemCreate {
  product_id: number;
  quantity: number;
}

export interface OrderCreate {
  customer_id: number;
  items: OrderItemCreate[];
}

export interface OrderItemResponse {
  id: number;
  product_id: number;
  quantity: number;
  purchased_price: DecimalString;
}

export interface OrderResponse {
  id: number;
  customer_id: number;
  items: OrderItemResponse[];
}

export type OrderListResponse = OrderResponse[];
export type OrderDeleteResponse = ApiMessageResponse;
